src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs

```diff
@@ -79,47 +79,43 @@ private static bool IsValidParameterList(IReadOnlyList<KeyValuePair<string, obje
         {
             hasMessageTemplateCapture = false;
             isMixedPositional = false;
-            isPositional = false;
             originalMessageIndex = null;
-            bool? firstParameterIsPositional = null;
+            isPositional = false;
+            string parameterName;
```

Comment:
Why is this one moved to the outer scope?

Reply:
Old c++ habbit. No need to push the same parameter on the stack for every loop.

Reply:
In my experience is could cause memory leaks.

Reply:
And I think it's a code smell in Sonar, not sure

Reply:
Only tried memory leaks in C# with dangling event-handlers (keeps things alive after lifetime) or dynamic-compiled-code (can never be freed).